### ISSUE TITLE
IDE0059 is not raised inside a code block with `using` keyword.

### ISSUE THREAD

**Issue by GillesTourreau** [dotnet/roslyn-analyzers#7339]
### Analyzer

**Diagnostic ID**: [IDE0059](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0059): `Remove unnecessary value assignment`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 8.0.302](https://dotnet.microsoft.com/en-us/download/dotnet/8.0)

### Describe the bug

IDE0059 rule is not raised inside a code block with the `using` keyword.

### Steps To Reproduce

Write the following code:

```csharp
public class Sandbox
{
    public void SimpleMethod()
    {
        using var _ = new MemoryStream();

        var jsonContent = JsonSerializer.Deserialize<Json>((JsonDocument)null);
    }

    public class Json
    {
    }
}
```

The `IDE0059` rule is not raised for the not used variable `jsonContent`.
But if we remove the `using` keyword:

```csharp
public class Sandbox
{
    public void SimpleMethod()
    {
        var _ = new MemoryStream();

        var jsonContent = JsonSerializer.Deserialize<Json>((JsonDocument)null);
    }

    public class Json
    {
    }
}
```

The `IDE0059` rule is raised to indicate that the `jsonContent` variable is not used.

We have the same issue if we use explicit code block for `using` keyword:

```csharp
public class Sandbox
{
    public void SimpleMethod()
    {
        using var _ = new MemoryStream();
        {
            var jsonContent = JsonSerializer.Deserialize<Json>((JsonDocument)null);
        }
    }

    public class Json
    {
    }
}
```

### Expected behavior

The `IDE0059` rule should be raised when a local variable is not used, even if there is a non-related `using` code block.

**Comment by GillesTourreau** [comment]
Sorry, wrong repository project. I created the [#74219](https://github.com/dotnet/roslyn/issues/74219) issue in the [dotnet/roselyn](https://github.com/dotnet/roslyn) project.
